### PR TITLE
Broker/Coordinator nodes start only after inventory is initialized

### PR DESCRIPTION
**Pull request by niketh**
Broker queries return wrong results just after startup because the inventory is not fully initialized and BrokerServerView is incomplete. The solution is to block `CuratorInventoryManager start()` method till inventory is fully available.

### CODE REVIEW

**Comment by himanshug** [comment]
:+1: ( @niketh  have you tested this ? )

**Comment by himanshug** [comment]
@niketh I don't think this will fix the problem, how will your clients know not to send requests before broker lifecycle has finished starting ?
With 0.11.0 , you should be able to get list of brokers from coordinator using `/druid/coordinator/v1/cluster/broker` endpoint , list returned there is populated only after a broker has initialized its inventory. Then your client can query that endpoint to get list of brokers which are ready.

**Comment by niketh** [comment]
@himanshug Tested in production and it works as expected. 

"how will your clients know not to send requests before broker lifecycle has finished starting ?"
>>Jetty will not start until the full view is loaded

File: server/src/main/java/io/druid/curator/inventory/CuratorInventoryManager.java [hunk 1]
```diff
@@ -122,6 +122,12 @@ public void start() throws Exception
       }
       throw e;
     }
+
+    // Block start() till PathChildrenCache is fully loaded
+    while (!containerCacheListener.doneInitializing) {
```
Comment (himanshug) [inline server/src/main/java/io/druid/curator/inventory/CuratorInventoryManager.java]: can we use a CountDownLatch.await(1 minute) here instead? that way not too many logs will be printed and also it wouldn't wait any more than necessary ?

File: server/src/main/java/io/druid/curator/inventory/CuratorInventoryManager.java [hunk 2]
```diff
@@ -69,6 +70,8 @@
 
   private volatile PathChildrenCache childrenCache;
 
+  private final CountDownLatch latch = new CountDownLatch(1);
```
Comment (himanshug) [inline server/src/main/java/io/druid/curator/inventory/CuratorInventoryManager.java]: can we call this `initializationDoneLatch` or something like that ? also , I think `doneInitializing` boolean variable can be removed and `latch.await(1 ms)` can be used to do the boolean check wherever necessary.

**Comment by gianm** [comment]
Same idea as #6742.

**Pull request closed without merging**